### PR TITLE
[C-Api/Service] internal function for ml-service handle

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Code clean, prepare new API for ml-service extension.
Add internal function for ml-service handle.

### CODE REVIEW

File: c/src/ml-api-service-agent-client.c [hunk 1]
```diff
@@ -395,31 +395,28 @@ ml_service_launch_pipeline (const char *name, ml_service_h * handle)
...
-  mls = g_try_new0 (ml_service_s, 1);
+  mls = _ml_service_create_internal (ML_SERVICE_TYPE_SERVER_PIPELINE);
   if (mls == NULL) {
     _ml_error_report_return (ML_ERROR_OUT_OF_MEMORY,
         "Failed to allocate memory for the service handle. Out of memory?");
   }
 
-  server = g_try_new0 (_ml_service_server_s, 1);
+  mls->priv = server = g_try_new0 (_ml_service_server_s, 1);
   if (server == NULL) {
```
Comment (songgot) [inline c/src/ml-api-service-agent-client.c:405]: Does ```mls->priv``` have a valid pointer before ```server``` is allocated?

File: c/src/ml-api-service-agent-client.c [hunk 2]
```diff
@@ -395,31 +395,28 @@ ml_service_launch_pipeline (const char *name, ml_service_h * handle)
...
+    _ml_error_report_return (ret,
+        "Failed to invoke the method launch_pipeline (%s).",
         (err ? err->message : "Unknown error"));
-    return ret;
   }
 
   server->service_name = g_strdup (name);
-  mls->type = ML_SERVICE_TYPE_SERVER_PIPELINE;
-  mls->priv = server;
   *handle = mls;
```
Comment (songgot) [inline c/src/ml-api-service-agent-client.c:420]: How about ```mls->priv = server``` in here?

File: c/src/ml-api-service-common.c [hunk 3]
```diff
@@ -13,37 +13,84 @@
...
-      ret = ml_service_pipeline_release_internal (mls->priv);
+      ret = ml_service_pipeline_release_internal (mls);
       break;
     case ML_SERVICE_TYPE_CLIENT_QUERY:
-      ret = ml_service_query_release_internal (mls->priv);
+      ret = ml_service_query_release_internal (mls);
       break;
     case ML_SERVICE_TYPE_REMOTE:
-      ret = ml_service_remote_release_internal (mls->priv);
+      ret = ml_service_remote_release_internal (mls);
```
Comment (songgot) [inline c/src/ml-api-service-common.c:72]: When calling ```_ml_service_destroy_internal()```, when ret is an error, I wonder if there is a need to handle the error in the part where the function is called.

File: c/src/ml-api-service-agent-client.c (hunk 2 above)
Comment (jaeyun-jung) [reply c/src/ml-api-service-agent-client.c:420]: It is same mls->priv = server = new0().. at line 404.

File: c/src/ml-api-service-common.c (hunk 3 above)
Comment (jaeyun-jung) [reply c/src/ml-api-service-common.c:72]: I will add log messge if ret != ML_ERROR_NONR.

Comment (jaeyun-jung) [reply c/src/ml-api-service-common.c:72]: updated

**Review by songgot — APPROVED** [review]
LGTM

**Review by wooksong — APPROVED** [review]

**Review by gichan-jang — APPROVED** [review]

**Pull request merged**